confmap/resolver_test.go

```diff
@@ -423,6 +423,45 @@ func TestResolverExpandMapAndSliceValues(t *testing.T) {
 	assert.Equal(t, expectedMap, cfgMap.ToStringMap())
 }
 
+func TestResolverExpandStringValues(t *testing.T) {
+	provider := newFakeProvider("input", func(context.Context, string, WatcherFunc) (*Retrieved, error) {
+		return NewRetrieved(map[string]any{
+			"test_no_match_old":             "${HOST}:${PORT}",
+			"test_no_match_old_no_brackets": "${HOST}:$PORT",
+			"test_match_value":              "${env:COMPLEX_VALUE}",
+			"test_match_embedded":           "${env:HOST}:3043",
+			"test_match_embedded_multi":     "${env:HOST}:${env:PORT}",
+		})
```

Comment:
Should we add a test case with a literal both at the beginning and at the end of the string? Say, `http://${env:HOST}:3043`

Reply:
Done.